### ISSUE TITLE
Should `\process` in a macro consume the input?

### ISSUE THREAD

**Issue by alerque** [sile-typesetter/sile#1006]
See https://gist.github.com/ctrlcctrlv/1ff9340d027cb206246b738845fb81c8#gistcomment-3400623

As a side effect of legitimate fixes to other issues (thanks @OlivierNicole) #863 and #910 lifted something that is documented as a limitation: namely that `\process` can only be called once in a given macro. The reason that was the case had to do with a buggy implementation and the current one both works better and doesn't add that limitation.

That being said I'm not convinced it's a good thing. This might open the door to a host of side-effect bugs. Or maybe not. We need to review what happens to the AST and what content may or may not break by processing it twice in the same document. Given the goal of the macro language in might be okay to explicitly block this.

If on the other hand the current behavior is for keeps then the documentation needs updating.

**Comment by Omikhleia** [comment]
> If on the other hand the current behavior is for keeps then the documentation needs updating.

The issue has been inactive for 4+ years, and the question of multiple invocations of `\process` doesn't seem to have re-occurred = I'd be in favor of the above suggestion, i.e. document properly the current behavior and then close this issue.

**Comment by alerque** [comment]
I agree, this hasn't been a problem and from my current vantage point, isn't likely to cause one either. Especially noted this was noted well *before* I even worked on the formal grammar (late 2022) much less convinced myself the grammar was sane and released the definition (early 2024). Given it wasn't a problem in 4 years and has no known negative ramifications, we should just drop the documentation of the non-existent limitation and let it go.
